core/trino-main/src/main/java/io/trino/connector/ConnectorAwareNodeManager.java

```diff
@@ -39,16 +40,15 @@ public ConnectorAwareNodeManager(InternalNodeManager nodeManager, String environ
     @Override
     public Set<Node> getAllNodes()
     {
-        return ImmutableSet.<Node>builder()
-                .addAll(getWorkerNodes())
-                .addAll(nodeManager.getCoordinators())
-                .build();
+        return ImmutableSet.copyOf(nodeManager.getActiveConnectorNodes(catalogName));
```

Comment:
I think the change here is not necessary, since coordinator (previously unconditionally returned here) _must_ host the catalog

Reply:
Coordinator is part of `getActiveConnectorNodes`. Hence `getAllNodes` just calls `getActiveConnectorNodes` (it doesn't need to be added).